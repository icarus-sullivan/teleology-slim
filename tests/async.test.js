const { slimAsync } = require('../src');

jest.setTimeout(15000);

const context = {
  location: async () => Promise.resolve('world'),
  what: 'slim',
};

const template = `
hello {{ location }}

this is a test script for {{ what }}`;

test('Async template is created', async () => {
  const result = await slimAsync(template, context);
  expect(result).toMatchSnapshot();
});

test('Missing value does not cause issues', async () => {
  const result = await slimAsync(template, {});
  expect(result).toMatchSnapshot();
});
