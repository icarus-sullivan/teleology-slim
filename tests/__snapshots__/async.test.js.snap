// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Async template is created 1`] = `
"
hello world

this is a test script for slim"
`;

exports[`Missing value does not cause issues 1`] = `
"
hello 

this is a test script for "
`;
